### ISSUE TITLE
a11y: Instruction not associated with form field

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#13829]
a11y issue 191

**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**

The mentioned instruction is not associated with its respective form field which resulting in screen reader users might miss out important instruction.

Repository: Create
- Will default to "master" if left blank (which gets available on selecting "Git repository containing Helm chart or cluster template definitions" radio button)

Users Retention Settings
- This value must be larger than the Disable if it's active

User: Create
  - Global Permissions and Built-in sections
  - Administrators have full control over the entire installation and all resources in all clusters. 
  - User-Base users have login-access only
  - Allows the user to enable, configure, and disable all Authentication provider settings
  - Allows the user to enable and disable custom features via feature flag settings and so on 

Apply the following changes:
• Add "aria-describedby" attribute to the form field.
• Reference the "id" value of the element used to display the instruction text in the "aria-describedby" attribute.

**To Reproduce**

**Result**

**Expected Result**

The screen reader should announce the instruction along with label text.

**Screenshots**

**Additional context**

**Comment by izaac** [comment]
Rancher version: `v2.12-7ca04114a589de3edf62e85eded0ba89c408ec2c-head`
Rancher UI version (dashboard): `master 28636c1`

All the UI elements mentioned in the issue have their associated description properly converted from text to speech matching their purpose clearly.

This build was compared to`v2.9.10-alpha1` for reference.

✅
